src/firewall.py

```diff
@@ -66,7 +68,38 @@ def get_host_ip(self) -> str:
         )
         return str(ipaddress.IPv4Interface(address.strip()).ip)
 
-    def refresh_firewall(self, denylist: typing.List[FirewallEntry]):
+    def _exclude_network(
+        self,
+        networks: typing.List[NetworkT],
+        excludes: typing.List[NetworkT],
+    ) -> typing.Set[NetworkT]:
+        """Excludes the network segment from a pool of networks.
+
+        Args:
+            networks: The network pool to apply.
+            excludes: The networks to exclude from the pool.
+
+        Returns:
+            The network pool without the network segments in excludes.
+        """
+        total_excluded_networks = set(networks)
```

Comment:
I would prefer something like `total_networks_without_excluded`, because one could think you would return the excluded networks.

Reply:
That is indeed a much better var name choice, thank you!